src/core/utils/hooks/useOverflow.tsx

```diff
@@ -60,21 +69,26 @@ export const useOverflow = <T extends HTMLElement>(
     const availableWidth = containerRef.current.offsetWidth;
     const requiredWidth = containerRef.current.scrollWidth;
 
-    // hide items when there's no space available
-    if (availableWidth < requiredWidth && visibleCount > 1) {
-      setVisibleCount((count) => count - 1);
-      overflowBreakpoints.current.push(availableWidth);
-    }
-    // restore items when there's enough space again
-    else if (
-      overflowBreakpoints.current.length > 0 &&
-      availableWidth >
-        overflowBreakpoints.current[overflowBreakpoints.current.length - 1]
-    ) {
-      setVisibleCount((count) => count + 1);
-      overflowBreakpoints.current.pop();
+    if (availableWidth < requiredWidth) {
+      const avgItemWidth = requiredWidth / visibleCount;
+      const visibleItems = Math.floor(availableWidth / avgItemWidth);
+      setVisibleCount(visibleItems);
+    } else if (needsFullRerender.current) {
+      const childrenWidth = Array.from(containerRef.current.children).reduce(
+        (sum: number, child: HTMLElement) => sum + child.offsetWidth,
+        0,
+      );
+      const avgItemWidth = childrenWidth / visibleCount;
```

Comment:
Why not use `requiredWidth`?

Reply:
If container fits 50 items and in the beginning I add 20 then single item width will be way too big because `offsetWidth` and `scrollWidth` should be all the posible container width.